libs/core/src/scss/interaction-state/_hover.scss

```diff
@@ -0,0 +1,54 @@
+@use 'sass:color';
+@use 'sass:meta';
+@use '../utils';
+
+// TODO: Accessibility - Loudness must respect user preferences
+// See https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast
+// TODO: Await UX decision on default loudness and update
+$default-loudness: utils.get-loudness('loud');
+
+@mixin apply-hover($background-color, $color, $property, $loudness: null) {
+  @if meta.type-of($background-color) != color {
+    @error "$background-color must be a valid CSS color value. See https://developer.mozilla.org/en-US/docs/Web/CSS/color_value";
+  }
+
+  @if meta.type-of($color) != color {
+    @error "$color must be a valid CSS color value. See https://developer.mozilla.org/en-US/docs/Web/CSS/color_value";
+  }
+
+  // ---------------------------------------------------------------------------
+  // Determine if lightness should increase or decrease.
+  // The default is increase (a positive percentage).
+  //
+  // When background color is transparent assume that the canvas underneath is
+  // a light color. We cannot really determine if lightness should increase or
+  // decrease, but this is a best guess. It's possible to work around by passing
+  // a $background-color (or $color) parameter with a value other than
+  // transparent.
+  //
+  // Sidenote: The color transparent is defined as black with 0% opacity, i.e.,
+  // 100% transparent. Often declared as rgba(0, 0, 0, 0%) in browsers.
+  // ---------------------------------------------------------------------------
+  $relative-lightness: $default-loudness !default;
+
+  @if $loudness {
+    $relative-lightness: $loudness;
+  }
```

Comment:
If we can eliminate the use of `null` - we can probably let go of this if-clause 😄

Reply:
Initiatially it was implemented without this condition, but I had to change it to accommodate different use cases.

I'll try to see if I can come up with something else, but I'm not sure there's another solution that covers everything.

If you have any other suggestions I'm all ears 👂🏻

Reply:
As a sidenote, one of the recommended ways to make function/mixin arguments optional is to assign `null` as default value. However, in this case we _do_ have a reasonable default value we can assign instead.

Maybe there is a problem in having two mixins `_button-hover()` and `apply-hover()` that both need to handle:

1. `$loudness` has been assigned a proper value
1. `$loudness` has been assigned to `null`
1. `$loudness` has not been assigned

but I'm not completely sure I have a better solution.

I can move the default assignment to `_button-hover()` to ensure that we are always invoking `apply-hover()` with a proper $loudness argument. But there may be cases where we don't want to do that.

Btw, assigning `null` as argument value in Sass is not the same as leaving the argument out. The latter will make the preprocessor fail with an error.

I'll keep digging.